src/pal/src/misc/dactableaddress.cpp

```diff
@@ -0,0 +1,156 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+/*++
+
+
+
+Module Name:
+
+    dactableaddress.cpp
+
+Abstract:
+
+    Functions to write and read DAC global pointer table address between the coreclr
+    and DAC/debugger processes.
+
+Revision History:
+
+
+--*/
+
+#include "pal/palinternal.h"
+#include <sys/types.h>
+#include <unistd.h>
+#include <stdio.h>
+
+/*++
+Function
+    PAL_PublishDacTableAddress
+
+Parameters
+
+    address
+       [in] address of dac table
+    size
+       [in] size of dac table
+
+Return Values
+       pal errors
+
+--*/
+PALIMPORT
+DWORD
+PALAPI
+PAL_PublishDacTableAddress(
+    IN PVOID address,
+    IN ULONG size)
+{
+    DWORD ret = NO_ERROR;
+
+    // TODO - 3/5/15 - the DAC side needs the debuggee pid
+    // pid_t pid = getpid();
+    pid_t pid = 0;
+    char fileName[100]; 
+    snprintf(fileName, sizeof(fileName), "/tmp/%d_dacTable", pid);
```

Comment:
This is temporary. Another function to get the dac table needs to be added to the ICLRDataTarget (actually create ICLRDataTarget2 and add it to that) that is passed to the DAC from the debugger  or DBI code.  For live and dump debugging it would be implemented with lldb API’s to get the g_dacTable symbol in the debuggee process. There would be no file involved.  For DBI, it would be more complicated since there is no native debugger involved. The new interface would have to implemented some other way TBD.

mikem

From: Jan Kotas [mailto:notifications@github.com]
Sent: Thursday, March 5, 2015 11:11 AM
To: dotnet/coreclr
Cc: Mike McLaughlin
Subject: Re: [coreclr] Implement the DAC global pointer table for Linux (#400)


In src/pal/src/misc/dactableaddress.cpp<https://github.com/dotnet/coreclr/pull/400#discussion_r25891758>:

> +

> +--*/

> +PALIMPORT

> +DWORD

> +PALAPI

> +PAL_PublishDacTableAddress(

> +    IN PVOID address,

> +    IN ULONG size)

> +{

> +    DWORD ret = NO_ERROR;

> +

> +    // TODO - 3/5/15 - the DAC side needs the debuggee pid

> +    // pid_t pid = getpid();

> +    pid_t pid = 0;

> +    char fileName[100];

> +    snprintf(fileName, sizeof(fileName), "/tmp/%d_dacTable", pid);

How is this going to work with dump debugging?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/400/files#r25891758>.